packages/medusa/src/types/pricing.ts

```diff
@@ -39,6 +31,11 @@ export type ShippingOptionPricing = {
   tax_amount: number
 }
 
+export type VariantData = {
+  id: string
+  product_id: string
+}
```

Comment:
you mean just passing the id and not the product_id? If that the case, it is the same answer as above. Accumulating a judge number of queries on top of each other to get the same data as we already hold will only increase the idle transaction time and response time. It is a trade-off and I can do all the changes you've asked and re do a benchmark. I think we will still be better than what we use to have. But, a lot higher than what we can get but let see

Reply:
my point was more to fetch the product variant within the method and then use `variant.product_id` instead of adding the responsibility to the caller of making sure that `id` and `product_id` are correctly associated with each other. I realize this will again introduce additional DB queries, but similar to the argument made previously it might be worth it to avoid complexity in the API, wdyt?

Reply:
I ll try it out and share the result soon 👍

Reply:
I ve done the changes in a bulk way. The perf are a bit better or a bit worst but still as good 🚀